jina/serve/runtimes/worker/batch_queue.py

```diff
@@ -97,73 +99,143 @@ async def push(self, request: DataRequest) -> Task:
                 self._flush_task = asyncio.create_task(self._await_then_flush())
             if not self._timer_task:
                 self._start_timer()
-
             self._big_doc.extend(docs)
-            self._requests.append(request)
+            nex_req_idx = len(self._requests)
+            num_docs = len(docs)
+            self._request_idxs.extend([nex_req_idx] * num_docs)
             self._request_lens.append(len(docs))
-
+            self._requests.append(request)
+            queue = asyncio.Queue()
+            self._requests_completed.append(queue)
             if len(self._big_doc) >= self._preferred_batch_size:
                 self._flush_trigger.set()
 
-        return self._flush_task
+        return queue
 
     async def _await_then_flush(self) -> None:
         """Process all requests in the queue once flush_trigger event is set."""
-        await self._flush_trigger.wait()
 
-        # writes to shared data between tasks need to be mutually exclusive
-        async with self._data_lock:
-            try:
+        def _get_buckets_completed_request_indexes(non_assigned_docs,
```

Comment:
It is not super clear to me what this method does. what does one "bucket" represent?